File: server/src/main/java/com/project/v1/entity/Poitem.java
package com.project.v1.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "poitem")
public class Poitem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "purchaseorder_id", nullable = false)
    private Purchaseorder purchaseorder;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "item_id", nullable = false)
    private Item item;

    @Column(name = "quantity", precision = 9, scale = 2)
    private BigDecimal quantity;

    @Column(name = "expectedlinetotal", precision = 12, scale = 2)
    private BigDecimal expectedlinetotal;

}

File: client/src/app/core/entity/poitem.ts
import {Item} from "./item";

export class POItem{

  id:number;
  item:Item;
  quantity:number;
  expectedlinetotal:number;

  constructor(id: number, item: Item, quantity: number, expectedlinetotal: number) {
    this.id = id;
    this.item = item;
    this.quantity = quantity;
    this.expectedlinetotal = expectedlinetotal;
  }

}

